tiledb/api/c_api/group/group_api_external.h

```diff
@@ -630,12 +632,28 @@ TILEDB_EXPORT capi_return_t tiledb_group_get_query_type(
  * @param recursive should we recurse into sub-groups
  * @return  `TILEDB_OK` for success and `TILEDB_ERR` for error.
  */
-TILEDB_EXPORT capi_return_t tiledb_group_dump_str(
+TILEDB_DEPRECATED_EXPORT capi_return_t tiledb_group_dump_str(
     tiledb_ctx_t* ctx,
     tiledb_group_t* group,
     char** dump_ascii,
     const uint8_t recursive) TILEDB_NOEXCEPT;
 
+/**
+ * Dump a string representation of a group.
+ *
```

Comment:
Missing example. The example _could_ highlight the user's responsibility to free `dump_ascii`.